RevPiDevice.c

```diff
@@ -243,7 +244,16 @@ int RevPiDevice_run(void)
 
 	// if the user-ioctl want to send a telegram, do it now
 	if (test_bit(REVPI_CORE_FLAG_PENDING_TELEGRAM, &piCore_g.flags)) {
-		piCore_g.statusUserTel = piIoComm_sendTelegram(&piCore_g.requestUserTel, &piCore_g.responseUserTel);
+		SIOGeneric *telegram = &piCore_g.requestUserTel;
+		UIoProtocolHeader *hdr = &telegram->uHeader;
+		char rcv_buff[IOPROTOCOL_MAXDATA_LENGTH];
+
+		piCore_g.statusUserTel = pibridge_req_io(hdr->sHeaderTyp1.bitAddress,
+							 hdr->sHeaderTyp1.bitCommand,
+							 telegram->ai8uData,
+							 hdr->sHeaderTyp1.bitLength,
+							 rcv_buff,
+							 sizeof(rcv_buff));
 		up(&piCore_g.semUserTel);
 	}
```

Comment:
The `KB_INTERN_IO_MSG` ioctl expects the response telegram in `piCore_g.responseUserTel`, yet you're only temporarily storing the response in `rcv_buff` on the stack. The stack gets thrown away when exiting the function.

I'd suggest passing `&piCore_g.responseUserTel.ai8uData` and `IOPROTOCOL_MAX_DATA_LENGTH` (or alternatively `sizeof(piCore_g.responseUserTel.ai8uData) - 1`) to `pibridge_req_io()`.

Reply:
The `- 1` is for the CRC in case that wasn't clear.

Reply:
AFAIK the only user of this function are we (KUNBUS). Do we really evaluate the returned Telegram? If not I would rather like to remove it from piCore_g.

Reply:
I would be open to dropping the `KB_INTERN_IO_MSG` ioctl and then the `piCore_g.responseUserTel` becomes obsolete as well. But it's probably reasonable to do that separately and after coordination with the PO & team, so retaining backward compatibility for now (as you've done) seems like a good idea.

Reply:
If I remember your explanations a few weeks ago this telegram type is (probably) is debug relic from our past? Then I'd be happy to drop this. Less code and more understood functionality hopefully increases the maintainability in the long run.

Reply:
Note, that the header of the response telegram does not contain any valid data. Should that also be adjusted to look like the header of a real response packet?